docs/decisions/0001-open-analytics.rst

```diff
@@ -0,0 +1,106 @@
+
+Decision
+********
+
+We will create the Open Analytics Reference System (OARS) that combines existing open-source projects
+into a preconfigured bundle that can be easily deployed using Tutor.
+
+These projects will include:
+
+- A Learning Record Store (LRS)
+- A service to transform tracking log events into an open standard
+- An analytic database
+- A data visualization and dashboard tool with a data export API
+- Code and configuration to tie these tools together, as well as rich reports that work against the
+  default configuration
+
+The guiding principals for technology selection are:
+
+- Based on open standards and open source
+- Hosting service agnostic
+- Inexpensive to run
+- Able to support near-real-time data where possible
+- Require little specialized knowledge to set up and maintain
+- Be extensible for a variety of common use cases not covered by the default configuration
+
+This system will:
+
+- Transform existing Open edX tracking log events into an open standard format
+- Store them using a standards-compliant learning record store
+- Present a user interface of data visualizations secured via single-sign-on against the LMS
+- Allow download of report data for those with permissions to view it
+- Provide a secure API for integrations with other tools or data viewing methods
+- Endeavor to be privacy preserving, by de-identifying learner data by default and focusing on
+  respecting learner privacy and data ownership when storing identity data
+
+Consequences
+************
+
+- Small and medium Open edX installs will have easy access to timely and relevant reports
+  about the usage of their site, the performance of their classes, and the status of their
+  students.
+- Use cases for advanced learner interventions and data-guided learning pathways will be
+  unblocked by access to near-real-time data provided in an industry standard format.
+- This reference implementation will replace Insights as the recommended analytics platform
+  for Open edX.
+
+Rejected Alternatives
+*********************
+
+Resurrect support of Insights
+-----------------------------
+Given the low adoption rate of Insights and its extremely high development, deployment, and
+maintenance costs, the value of this work seemed low.
+
+Rewrite Insights
+----------------
+A complete rewrite of the Insights project could have met many of our goals here, however
+our focus is on education. The cost and maintenance burden of creating a bespoke analytics
+pipeline and visualization solution is an unnecessary distraction when excellent open source
+tools exist that are much more feature rich, configurable, and better maintained than we could
+manage given our competing priorities.
+
+Use an existing community system
+--------------------------------
+In the absense of an officially supported analytics system, several organizations have created
+their own solutions. At the time of investigation none supported all of the features we are hoping
```

Comment:
What are the impediments to adding the missing features to the existing systems? Is that worth mentioning?

Reply:
They are fairly specific to the systems, but generally things like relying on the production transactional databases for data storage or directly using tracking logs that would require substantial rewrites of systems that were already in production and likely a full reload of data into the new formats. I don't think it adds much to call them out specifically.